packages/manager/apps/pci-users/src/pages/index/components/RolesMatrix.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+  OsdsButton,
+  OsdsIcon,
+  OsdsPopover,
+  OsdsPopoverContent,
+  OsdsText,
+  OsdsSelect,
+  OsdsSelectOption,
+} from '@ovhcloud/ods-components/react';
+import {
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_TEXT_ALIGN,
+  ODS_BUTTON_VARIANT,
+  ODS_ICON_NAME,
+  ODS_ICON_SIZE,
+  ODS_TEXT_COLOR_HUE,
+  OdsSelectValueChangeEventDetail,
+  OsdsSelectCustomEvent,
+} from '@ovhcloud/ods-components';
+import {
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_TYPOGRAPHY_LEVEL,
+  ODS_THEME_TYPOGRAPHY_SIZE,
+} from '@ovhcloud/ods-common-theming';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useMedia } from 'react-use';
+import { Role, Service } from '@/interface';
+import RolesSelector from '@/pages/add/components/RolesSelector';
+import { BreakPoints } from '@/layout';
+
+interface IProps {
+  roles: Role[];
+  services: Service[];
+}
+
+export default function RolesMatrix({ roles, services }: IProps) {
+  const { t } = useTranslation('common');
+
+  const [toggled, setToggled] = useState<boolean>(false);
+  const [mdRoles, setMdRoles] = useState<Role[]>(roles);
+  const [smRole, setSmRole] = useState<Role>(null);
+  const [uiRoles, setUiRoles] = useState<Role[]>(mdRoles);
+
+  const isScreenMdPlus = useMedia(`(min-width: ${BreakPoints.MD}px)`);
+
+  useEffect(() => {
+    setMdRoles(roles);
+  }, [roles]);
+
+  useEffect(() => {
+    setSmRole(null);
+  }, [isScreenMdPlus]);
+
+  useEffect(() => {
+    if (isScreenMdPlus) setUiRoles(mdRoles);
+    else setUiRoles([smRole]);
+  }, [smRole, mdRoles]);
+
+  return (
+    <>
+      <div>
+        <OsdsButton
+          size={ODS_BUTTON_SIZE.sm}
+          variant={ODS_BUTTON_VARIANT.stroked}
+          color={ODS_THEME_COLOR_INTENT.primary}
+          className="inline-block"
+          onClick={() => setToggled(!toggled)}
+        >
+          {toggled
+            ? t('pci_projects_project_users_matrix_hide')
+            : t('pci_projects_project_users_matrix_show')}
+        </OsdsButton>
+      </div>
+      {toggled && (
+        <div>
+          {isScreenMdPlus ? (
```

Comment:
since the jsx is quite heavy, could you split it in components ?
for example a RolesMatrixMobile / RolesMatrixDesktop

maybe it could also make state management easier for the roles

Reply:
on table level there is no distinction between mobile and desktop, only data changes, but i'll try to break the components into at least two separate components